editor/src/clj/editor/atlas.clj

```diff
@@ -67,28 +69,45 @@
 
 (vtx/defvertex texture-vtx
   (vec4 position)
-  (vec2 texcoord0))
+  (vec2 texcoord0)
+  (vec1 page_index))
 
 (shader/defshader pos-uv-vert
   (attribute vec4 position)
   (attribute vec2 texcoord0)
+  (attribute float page_index)
   (varying vec2 var_texcoord0)
+  (varying float var_page_index)
   (defn void main []
     (setq gl_Position (* gl_ModelViewProjectionMatrix position))
-    (setq var_texcoord0 texcoord0)))
+    (setq var_texcoord0 texcoord0)
+    (setq var_page_index page_index)))
 
-(shader/defshader pos-uv-frag
-  (varying vec2 var_texcoord0)
-  (uniform sampler2D texture_sampler)
-  (defn void main []
-    (setq gl_FragColor (texture2D texture_sampler var_texcoord0.xy))))
+(shader/defshader-with-array-samplers pos-uv-frag "texture_sampler"
+    (varying vec2 var_texcoord0)
+    (varying float var_page_index)
+    (defn void main []
+    (setq gl_FragColor (texture2DArray var_texcoord0.xy var_page_index))))
```

Comment:
Yeah hm, what's the reason we use that macro anyway? Is it because it's easier to edit in a code editor than a string? It feels like we are adding overhead with very little gain.. But yes, we could probably use BOB here.